api/all/src/main/java/io/opentelemetry/api/common/ExtendedAttributes.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.common;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@SuppressWarnings("rawtypes")
+@Immutable
+public interface ExtendedAttributes {
```

Comment:
How about renaming to `ComplexAttributes`?

 It does not suggest that there is some inheritance or that it e.g. supports more primitibive types.
I think that "complex" is a better term for a type which supports nesting. _I see that "complex" is even refered in https://refactoring.guru/design-patterns/composite._
At last the proposed name is shorter 😉

Reply:
(feel free to resolve my comment)